From 7b9f4b37edafec8d3a084b9bf2cbb35175a3155d Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Wed, 28 Dec 2022 01:01:41 +0800
Subject: [PATCH 24/28] Revert "drm/i915/sdvo: Filter out invalid outputs more
 sensibly"

This reverts commit b86830cc95af9bc0ba27041f163ce61a03bcded2.
---
 drivers/gpu/drm/i915/display/intel_sdvo.c | 27 +++++------------------
 1 file changed, 5 insertions(+), 22 deletions(-)

diff --git a/drivers/gpu/drm/i915/display/intel_sdvo.c b/drivers/gpu/drm/i915/display/intel_sdvo.c
index 2d22b7bb11ca5..4eaa4aa86ecdf 100644
--- a/drivers/gpu/drm/i915/display/intel_sdvo.c
+++ b/drivers/gpu/drm/i915/display/intel_sdvo.c
@@ -2937,33 +2937,16 @@ intel_sdvo_lvds_init(struct intel_sdvo *intel_sdvo, int device)
 	return false;
 }
 
-static u16 intel_sdvo_filter_output_flags(u16 flags)
-{
-	flags &= SDVO_OUTPUT_MASK;
-
-	/* SDVO requires XXX1 function may not exist unless it has XXX0 function.*/
-	if (!(flags & SDVO_OUTPUT_TMDS0))
-		flags &= ~SDVO_OUTPUT_TMDS1;
-
-	if (!(flags & SDVO_OUTPUT_RGB0))
-		flags &= ~SDVO_OUTPUT_RGB1;
-
-	if (!(flags & SDVO_OUTPUT_LVDS0))
-		flags &= ~SDVO_OUTPUT_LVDS1;
-
-	return flags;
-}
-
 static bool
 intel_sdvo_output_setup(struct intel_sdvo *intel_sdvo, u16 flags)
 {
-	flags = intel_sdvo_filter_output_flags(flags);
+	/* SDVO requires XXX1 function may not exist unless it has XXX0 function.*/
 
 	if (flags & SDVO_OUTPUT_TMDS0)
 		if (!intel_sdvo_dvi_init(intel_sdvo, 0))
 			return false;
 
-	if (flags & SDVO_OUTPUT_TMDS1)
+	if ((flags & SDVO_TMDS_MASK) == SDVO_TMDS_MASK)
 		if (!intel_sdvo_dvi_init(intel_sdvo, 1))
 			return false;
 
@@ -2984,7 +2967,7 @@ intel_sdvo_output_setup(struct intel_sdvo *intel_sdvo, u16 flags)
 		if (!intel_sdvo_analog_init(intel_sdvo, 0))
 			return false;
 
-	if (flags & SDVO_OUTPUT_RGB1)
+	if ((flags & SDVO_RGB_MASK) == SDVO_RGB_MASK)
 		if (!intel_sdvo_analog_init(intel_sdvo, 1))
 			return false;
 
@@ -2992,11 +2975,11 @@ intel_sdvo_output_setup(struct intel_sdvo *intel_sdvo, u16 flags)
 		if (!intel_sdvo_lvds_init(intel_sdvo, 0))
 			return false;
 
-	if (flags & SDVO_OUTPUT_LVDS1)
+	if ((flags & SDVO_LVDS_MASK) == SDVO_LVDS_MASK)
 		if (!intel_sdvo_lvds_init(intel_sdvo, 1))
 			return false;
 
-	if (flags == 0) {
+	if ((flags & SDVO_OUTPUT_MASK) == 0) {
 		unsigned char bytes[2];
 
 		intel_sdvo->controlled_output = 0;
-- 
2.25.1

